xen/drivers/passthrough/iommu.c

```diff
@@ -142,7 +142,14 @@ int iommu_domain_init(struct domain *d, bool_t use_iommu)
         return 0;
 
     hd->platform_ops = iommu_get_ops();
-    return hd->platform_ops->init(d);
+    ret = hd->platform_ops->init(d);
```

Comment:
done. Now this patch isn't present as a standalone one since I squashed two patches you had proposed to squash.
And the final commit message looks detailing enough:
    iommu: Pass additional use_iommu argument to iommu_domain_init()
    
    The presence of this flag lets us know that the guest
    has devices which will most likely be used for passthrough.
    In that case we have to call iommu_construct(), actually
    what the real assign_device call usually does.
    
    As iommu_domain_init() is called with forced to false use_iommu flag
    for now, no functional change is intended.
    
    Basically, this patch is needed for unshared IOMMUs on ARM only
    since the unshared IOMMUs on x86 are ok if iommu_construct() is called
    later. But, in order to be more generic and for possible future optimization
    make this change appliable for both platforms.
    So, the patch target is to make ARM happy and not to brake x86.
    Confirmation from x86 guys is needed.




 single patch that contains two old patches: